src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,92 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> components = new HashMap<>();
+
+    private Injector() {
+        try {
+            registerComponents();
+            injectDependencies();
+        } catch (Exception e) {
+            throw new RuntimeException("Error initializing injector", e);
+        }
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    private void registerComponents() {
+        Class<?>[] classes = {
+                ProductServiceImpl.class,
+                ProductParserImpl.class,
+                FileReaderServiceImpl.class
+        };
+
+        for (Class<?> clazz : classes) {
+            if (clazz.isAnnotationPresent(Component.class)) {
+                Constructor<?> constructor;
+                try {
+                    constructor = clazz.getDeclaredConstructor();
+                } catch (NoSuchMethodException e) {
+                    throw new RuntimeException("Could not find constructor for "
+                            + clazz.getName(), e);
+                }
+                Object instance = null;
+                try {
+                    instance = constructor.newInstance();
+                } catch (InstantiationException
+                         | IllegalAccessException
+                         | InvocationTargetException e) {
+                    throw new RuntimeException("Could not instantiate "
+                            + clazz.getName(), e);
```

Comment:
Similar to the previous comment, it is better to replace many exceptions that have a common parent with a general parental exception. In this case, `ReflectiveOperationException` can be used instead of listing all individual exceptions.